exercises/custom-set/custom_set_test.rb

```diff
@@ -3,114 +3,303 @@
   end
 
-  def test_put
+  def test_add_to_non_empty_set
     skip
-    assert_equal CustomSet.new([1, 2, 3, 4]),
-                 CustomSet.new([1, 2, 4]).put(3)
+    set = CustomSet.new [1, 2, 4]
+    element = 3
+    expected = CustomSet.new [1, 2, 3, 4]
+    assert_equal expected, set.put(element)
+  end
 
-    assert_equal CustomSet.new([1, 2, 3]),
-                 CustomSet.new([1, 2, 3]).put(3)
+  def test_adding_an_existing_element_does_not_change_the_set
+    skip
+    set = CustomSet.new [1, 2, 3]
+    element = 3
+    expected = CustomSet.new [1, 2, 3]
+    assert_equal expected, set.put(element)
+  end
 
-    assert_equal CustomSet.new([1, 2, 3, 3.0]),
-                 CustomSet.new([1, 2, 3]).put(3.0)
+  def test_intersection_of_two_empty_sets_is_an_empty_set
+    skip
+    set1 = CustomSet.new []
+    set2 = CustomSet.new []
+    expected = CustomSet.new []
+    assert_equal expected, set2.intersection(set1)
   end
 
-  def test_size
+  def test_intersection_of_an_empty_set_and_non_empty_set_is_an_empty_set
     skip
-    assert_equal 0, CustomSet.new.size
-    assert_equal 3, CustomSet.new([1, 2, 3]).size
-    assert_equal 3, CustomSet.new([1, 2, 3, 2]).size
+    set1 = CustomSet.new []
+    set2 = CustomSet.new [3, 2, 5]
+    expected = CustomSet.new []
+    assert_equal expected, set2.intersection(set1)
   end
 
-  def test_subset?
+  def test_intersection_of_a_non_empty_set_and_an_empty_set_is_an_empty_set
     skip
-    assert CustomSet.new([1, 2, 3]).subset?(CustomSet.new([1, 2, 3]))
-    assert CustomSet.new([4, 1, 2, 3]).subset?(CustomSet.new([1, 2, 3]))
-    refute CustomSet.new([4, 1, 3]).subset?(CustomSet.new([1, 2, 3]))
-    refute CustomSet.new([1, 2, 3, 4]).subset?(CustomSet.new([1, 2, 3.0]))
-    assert CustomSet.new([4, 1, 3]).subset?(CustomSet.new)
-    assert CustomSet.new.subset?(CustomSet.new)
+    set1 = CustomSet.new [1, 2, 3, 4]
+    set2 = CustomSet.new []
+    expected = CustomSet.new []
+    assert_equal expected, set2.intersection(set1)
   end
 
-  def test_to_a
+  def test_intersection_of_two_sets_with_no_shared_elements_is_an_empty_set
     skip
-    assert_equal [], CustomSet.new.to_a.sort
-    assert_equal [1, 2, 3], CustomSet.new([3, 1, 2]).to_a.sort
-    assert_equal [1, 2, 3], CustomSet.new([3, 1, 2, 1]).to_a.sort
+    set1 = CustomSet.new [1, 2, 3]
+    set2 = CustomSet.new [4, 5, 6]
+    expected = CustomSet.new []
+    assert_equal expected, set2.intersection(set1)
   end
 
-  def test_union # rubocop:disable Metrics/MethodLength
+  def test_intersection_of_two_sets_with_shared_elements_is_a_set_of_the_shared_elements
+    skip
+    set1 = CustomSet.new [1, 2, 3, 4]
+    set2 = CustomSet.new [3, 2, 5]
+    expected = CustomSet.new [2, 3]
+    assert_equal expected, set2.intersection(set1)
+  end
+
+  def test_difference_of_two_empty_sets_is_an_empty_set
+    skip
+    set1 = CustomSet.new []
+    set2 = CustomSet.new []
+    expected = CustomSet.new([])
```

Comment:
Noted, I will look in the cases and see if I added it in the method.  I wonder why Rubocop wouldn't have picked this up.